shotover-proxy/tests/opensearch_int_tests/mod.rs

```diff
@@ -87,3 +69,183 @@ async fn passthrough_standard() {
+//         .format("json")
+//         .pretty(true)
+//         .send()
+//         .await
+//         .unwrap();
+//
+//     assert_eq!(response.status_code(), StatusCode::OK);
+//     assert!(response
+//         .headers()
+//         .get(opensearch::http::headers::CONTENT_TYPE)
+//         .unwrap()
+//         .to_str()
+//         .unwrap()
+//         .starts_with("application/json"));
+//
+//     let response_body = response.json::<Value>().await.unwrap();
+//     assert_eq!(response_body[0]["status"], String::from("green"));
+// }
+
+async fn test_create_index(client: &OpenSearch) {
+    client
+        .indices()
+        .create(IndicesCreateParts::Index("test-index"))
+        .send()
+        .await
+        .unwrap();
+
+    let exists_response = client
+        .indices()
+        .exists(IndicesExistsParts::Index(&["test-index"]))
+        .send()
+        .await
+        .unwrap();
+
+    assert_eq!(exists_response.status_code(), StatusCode::OK);
+}
+
+async fn test_index_and_search_document(client: &OpenSearch) -> String {
+    client
+        .index(IndexParts::Index("test-index"))
+        .body(json!({
+            "name": "John",
+            "age": 30
+        }))
+        .refresh(Refresh::WaitFor)
+        .send()
+        .await
+        .unwrap();
+
+    let response = client
+        .search(SearchParts::Index(&["test-index"]))
+        .from(0)
+        .size(10)
+        .body(json!({
+            "query": {
+                "match": {
+                    "name": "John",
+                }
+            }
+        }))
+        .send()
+        .await
+        .unwrap();
+
+    let results = response.json::<Value>().await.unwrap();
+    assert!(results["took"].as_i64().is_some());
+    assert_eq!(
+        results["hits"].as_object().unwrap()["hits"]
+            .as_array()
+            .unwrap()
+            .len(),
+        1
+    );
+    results["hits"].as_object().unwrap()["hits"]
+        .as_array()
+        .unwrap()[0]["_id"]
+        .as_str()
+        .unwrap()
+        .to_string()
+}
+
+async fn test_delete_and_search_document(client: &OpenSearch, id: String) {
+    client
+        .delete(DeleteParts::IndexId("test-index", &id))
+        .refresh(Refresh::WaitFor)
+        .send()
+        .await
+        .unwrap();
+
+    let response = client
+        .search(SearchParts::Index(&["test-index"]))
+        .from(0)
+        .size(10)
+        .body(json!({
+            "query": {
+                "match": {
+                    "name": "John",
+                }
+            }
+        }))
+        .allow_no_indices(true)
+        .send()
+        .await
+        .unwrap();
+
+    let results = response.json::<Value>().await.unwrap();
+    assert!(results["took"].as_i64().is_some());
+    assert_eq!(
+        results["hits"].as_object().unwrap()["hits"]
+            .as_array()
+            .unwrap()
+            .len(),
+        0
+    );
+}
+
+async fn test_delete_index(client: &OpenSearch) {
```

Comment:
can we combine with test_create_index into a test_create_and_delete_index?
That way we are not at risk of breaking things by reordering or introducing new test cases.
e.g.
```rust
async fn test_create_and_delete_index(client: &OpenSearch) {
    client
        .indices()
        .create(IndicesCreateParts::Index("test-index"))
        .send()
        .await
        .unwrap();

    let exists_response = client
        .indices()
        .exists(IndicesExistsParts::Index(&["test-index"]))
        .send()
        .await
        .unwrap();

    assert_eq!(exists_response.status_code(), StatusCode::OK);
    
    client
        .indices()
        .delete(IndicesDeleteParts::Index(&["test-index"]))
        .send()
        .await
        .unwrap();

    let exists_response = client
        .indices()
        .exists(IndicesExistsParts::Index(&["test-index"]))
        .send()
        .await
        .unwrap();

    assert_eq!(exists_response.status_code(), StatusCode::NOT_FOUND);
}
```